src/java.base/share/classes/jdk/internal/util/ByteArrayLittleEndian.java

```diff
@@ -0,0 +1,424 @@
+     *                                   the range [0, array.length - 2]
+     * @see #getFloatRaw(byte[], int)
+     */
+    public static void setFloatRaw(byte[] array, int offset, float value) {
+        // Just sets the bits as they are
+        FLOAT.set(array, offset, value);
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using little endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 4]
+     * @see #getLong(byte[], int)
+     */
+    public static void setLong(byte[] array, int offset, long value) {
+        LONG.set(array, offset, value);
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using little endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * Variants of {@linkplain Double#NaN } values are canonized to a single NaN value.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 2]
+     * @see #getDouble(byte[], int)
+     */
+    public static void setDouble(byte[] array, int offset, double value) {
+        // Using Double.doubleToLongBits collapses NaN values to a single
+        // "canonical" NaN value
+        LONG.set(array, offset, Double.doubleToLongBits(value));
+    }
+
+    /**
+     * Sets (writes) the provided {@code value} using little endian order into
+     * the provided {@code array} beginning at the given {@code offset}.
+     * <p>
+     * Variants of {@linkplain Double#NaN } values are silently written according to
+     * their bit patterns.
+     * <p>
+     * There are no access alignment requirements.
+     *
+     * @param array  to set (write) a value into
+     * @param offset where setting (writing) in the array should begin
+     * @param value  value to set in the array
+     * @throws IndexOutOfBoundsException if the provided {@code offset} is outside
+     *                                   the range [0, array.length - 2]
+     * @see #getDoubleRaw(byte[], int)
+     */
+    public static void setDoubleRaw(byte[] array, int offset, double value) {
+        // Just sets the bits as they are
+        DOUBLE.set(array, offset, value);
+    }
+
+    private static VarHandle create(Class<?> viewArrayClass) {
```

Comment:
Renaming the method to `createLittleEndian` would make the static assignments at the top unambiguous.